### ISSUE TITLE
Improve HTTP response handling

### ISSUE THREAD

**Issue by MartinKolarik** [jsdelivr/globalping#347]
We have three main fields in the HTTP responses:
 - rawHeaders (HTTP specific)
 - rawBody (HTTP specific)
 - rawOutput (common for all test types)

The first two are clear and work well, but all tools are going to use the `rawOutput` field by default. The current behavior of `rawOutput` is:
 - copy `rawHeaders` if the request method was HEAD,
 - copy `rawBody` if the request method was GET.

In the second case, there's a problem because the body may be empty in some cases, most often redirects and error responses, but in general, also for normal responses. This makes the default output of tools relying on `rawOutput` look broken/confusing in those cases. @xbpcb hit this issue now while working on the website, and the CLI and I expect the bots to have the same issue. 

The current behavior matches that of `curl` when run without any switches but in my opinion, it would be better to make it closer to `curl -i` and have `rawOutput` have the full response (headers + body with one empty line in between, same as the real response). It's called "raw output" after all, and this is more useful for typical debugging use cases in most environments. Tools can still access the `rawBody` property if needed.

Additionally, the CLI should follow the approach that curl uses, and print the headers to `stderr` and the body to `stdout` to make the piping of the output work as before while proving both when output is not piped.

In other words, the proposed changes:
1. `rawOutput` returned from the API is changed to always include the status line, the headers, and the body, making the output more useful for tools that don't implement special handling of HTTP.
2. All our tools except the CLI keep using `rawOutput` as they do now and don't change anything, which means they show the headers for both request types.
3. The CLI is updated to output headers to `stderr` and body to `stdout`. This matches how `curl -i` works and provides more information while still making it possible to pipe the body to a file.

**Comment by jimaek** [comment]
The CLI is blocked from new releases until this is fixed so lets prioritize it

**Comment by alexey-yarmosh** [comment]
Should we also update `rawOutput` to include status and headers in case of errors? E.g.:
https://github.com/jsdelivr/globalping-probe/blob/4d5a6e2003c78a683078d7be850b1e6aa8da0004/test/unit/command/http-command.test.ts#L917
https://github.com/jsdelivr/globalping-probe/blob/4d5a6e2003c78a683078d7be850b1e6aa8da0004/test/unit/command/http-command.test.ts#L999

**Comment by MartinKolarik** [comment]
In case of HTTP errors yes, other errors no.
